web/js/codeworld-mode.js

```diff
@@ -82,43 +116,91 @@ CodeMirror.defineMode('codeworld', (_config, modeConfig) => {
 
         if (stream.match(RE_DASHES)) {
             stream.skipToEnd();
-            return 'comment';
+            return ['comment'];
         }
 
-        if (stream.match(RE_QUAL)) return 'qualifier';
-        if (stream.match(RE_VARID) || stream.match(RE_VARSYM)) return 'variable';
-        if (stream.match(RE_CONID) || stream.match(RE_CONSYM)) return 'variable-2';
-        if (stream.match(RE_NUMBER)) return 'number';
-        if (stream.match(RE_CHAR) || stream.match(RE_STRING)) return 'string';
+        if (checkContextStarter(stream, state)) {
+            return ['keyword'];
+        }
+
+        if (state.scanState === NORMAL && stream.match(RE_IN)) {
+            // Cancel previous layout 'let' context
+            state.layoutContext.pop();
+            return ['keyword'];
+        } else if (state.scanState === NORMAL) {
+            let lastContext = state.layoutContext[state.layoutContext.length - 1];
+            lastContext = lastContext === '{' ? 0 : lastContext;
+            if (stream.sol() && stream.indentation() < lastContext) {
```

Comment:
I'm not clear on what sol() means here.  Will it ignore whitespace?  Will it ignore comments?

For that matter, why do you need it at all?  If your current column is less than the latest layout context, you must have started a new line, right?

Reply:
Yes, it is leftover I forgot to delete.